test/integration/src/test/java/io/pravega/test/integration/KVTableWithStreamTest.java

```diff
@@ -0,0 +1,769 @@
+
+            log.info("Creating writer");
+            @Cleanup
+            final EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                    new JavaSerializer<>(), EventWriterConfig.builder().build());
+
+            // start writing events to the stream.
+            for (int i = 0; i < numEvents; i++) {
+                log.info("Writing event {}", streamEvent);
+                writer.writeEvent(routingKey, streamEvent);
+                writeCount++;
+                writer.flush();
+            }
+            log.info("Closing writer {}", writer);
+            writer.close();
+
+            // create a reader group.
+            log.info("Creating Reader group : {}", readerGroupName);
+
+            readerGroupManager.createReaderGroup(readerGroupName, ReaderGroupConfig.builder()
+                    .stream(Stream.of(newScope, streamName)).build());
+            log.info("Reader group name {} ", readerGroupManager.getReaderGroup(readerGroupName).getGroupName());
+            log.info("Reader group scope {}", readerGroupManager.getReaderGroup(readerGroupName).getScope());
+
+            log.info("Creating reader");
+            @Cleanup
+            final EventStreamReader<String> reader = clientFactory.createReader(readerName,
+                    readerGroupName,
+                    new JavaSerializer<>(),
+                    ReaderConfig.builder().build());
+
+            // start reading events from the stream.
+            while (!(readCount == writeCount)) {
+                final String eventRead = reader.readNextEvent(SECONDS.toMillis(2)).getEvent();
+                log.info("Reading event {}", eventRead);
+                if (eventRead != null) {
+                    // update if event read is not null.
+                    readCount++;
+                }
+            }
+            log.info("Closing reader {}", reader);
+            reader.close();
+
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        log.info("All writers have stopped. Event Written Count:{}, Event Read " +
+                "Count: {}", writeCount, readCount);
+        assertEquals("Number of events written & read should be equal", writeCount, readCount);
+
+        log.info("Read write test on stream succeeds");
+
+        KeyValueTableFactory newKeyValueTableFactory = new KeyValueTableFactoryImpl(newScope, this.controller, this.connectionPool);
+        KeyValueTable<Integer, String> newKeyValueTable = newKeyValueTableFactory.forKeyValueTable(newKVT.getKeyValueTableName(),
+                KEY_SERIALIZER, VALUE_SERIALIZER, KeyValueTableClientConfiguration.builder().build());
+
+        String keyFamily = "KeyFamily1";
+        LinkedHashMap<Integer, String> entryMap1 = new LinkedHashMap<>();
+        Integer[] keyArray = new Integer[10];
```

Comment:
You can combine these 3 loops in 1.

Reply:
Sure, I'll refactor the code here. Thanks